dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -422,7 +422,7 @@ def median(self) -> float:
         :rtype: float
         """
         if not self._has_histogram or not self._median_is_enabled:
-            return np.nan
+            return cast(float, np.nan)
```

Comment:
I couldn't figure out why `np.nan` wasn't being typed as `float` instead of `Any`. I thought maybe mypy wasn't using numpy's type stubs but apparently type stubs come by default in recent versions of numpy (like the one in our `requirements.txt`). 

Maybe this is caused by my local setup or the project's setup so if someone could confirm if reverting this line causes a pre-commit error, that would be great.

Reply:
recreated the error on my end when taking out the `cast(float...)`

```
dataprofiler/profilers/numerical_column_stats.py:425: error: Returning Any from function declared to return "float"
Found 1 error in 1 file (checked 55 source files)
```

Reply:
Should we update the return type then instead of casting? Numpy does typing well, how do they handle it?

Reply:
Turns out the reason for this was that we needed to add numpy as a dependency to the mypy pre-commit hook. This gave mypy the necessary typing information to fix this issue. I just made a commit that fixes this and removes some extraneous casting. [0ad2a38](https://github.com/capitalone/DataProfiler/pull/703/commits/0ad2a38a34143307d51a127ccd472d387f46864d)

Reply:
does this remove any other need for `cast()` in the other parts of the PR?

Reply:
Potentially, though I didn't see any other areas we could've removed it. A decent amount of casting is just due to the numpy's methods' return types being ambiguous when in reality we know what the type is.

Reply:
gotcha -- we should probably create a ticket around cleaning these up in the future IMO